### PR TITLE
Adding support for RTA_VIA

### PR DESCRIPTION
**Pull request by fach**
RTA_VIA is the netlink attribute type used to specify a gateway for a route using a different address family than said route. An example of this is using a IPv4 next hop for an IPv6 route. This PR adds support to set/get routes containing the RTA netlink attribute type.

Signed-off-by: Steve Shaw <shaw38@gmail.com>

### CODE REVIEW

**Comment by aboch** [comment]
Can you please add or extend existing UT functions to test the programming and reading of the `via` setting

File: route_test.go [hunk 1]
```diff
@@ -1339,3 +1340,68 @@ func TestMTURouteAddDel(t *testing.T) {
...
+	}
+	routes, err = RouteList(link, FAMILY_V4)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(routes) != 0 {
+		t.Fatal("Route not removed properly")
+	}
+
+	tearDown()
```
Comment (aboch) [inline route_test.go:1406]: any reason for not to using `defer` ?

Comment (fach) [reply route_test.go]: Nope, good catch.

File: route_test.go [hunk 2]
```diff
@@ -1339,3 +1340,67 @@ func TestMTURouteAddDel(t *testing.T) {
...
+
+	routes, err := RouteList(link, FAMILY_V4)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(routes) != 1 {
+		t.Fatal("Route not added properly")
+	}
+
+	if !cmp.Equal(route.MultiPath[0].Via, routes[0].Via) {
```
Comment (aboch) [inline route_test.go:1391]: do we really need to import this go-cmp/cmp to compare a two fields structure also considering the structure already has a `Equal()` method? 

Unless widely used and needed, I'd prefer to keep this project lean from outside imports

Comment (fach) [reply route_test.go]: Nope, it can be removed.

**Comment by aboch** [comment]
Thanks @fach please squash your commits into one and force push.

**Comment by fach** [comment]
Tests were failing due to the kernel version on the test runner. Added minimum kernel version to the unit tests for this.

**Pull request merged**